yao/benchmarks.jl

```diff
@@ -25,6 +25,16 @@ function build_circuit(n, nlayers, pairs)
     return circuit
 end
 
+qft_rotation(i, j) = control(i, j=>shift(2π/(1<<(i-j+1))))
+qft_layer(n, k) = chain(n, j==k ? put(k=>H) : qft_rotation(j, k) for j in k:n)
+qft(n) = chain(qft_layer(n, k) for k in 1:n)
+
+function build_qft_circuit(n)
+    circuit = chain(n)
+    push!(circuit, qft(n))
```

Comment:
```suggestion
qft(n)
```

you don't need to create a `chain`, any "block" can be executed.